src/harness/testcases/WINNF_FT_S_EXZ_testcase.py

```diff
@@ -0,0 +1,315 @@
+        'lowFrequency'] = 3625000000.0
+    grant_2['operationParam']['operationFrequencyRange'][
+        'highFrequency'] = 3635000000.0
+
+    # Create the actual config.
+    devices = [device_a, device_b, device_c]
+    grants = [grant_0, grant_1, grant_2]
+    config = {
+        'fccIds': [(d['fccId'], 47) for d in devices],
+        'userIds': [d['userId'] for d in devices],
+        'registrationRequests': devices,
+        'expectedResponseCodes': [(0,), (0,), (0,)],
+        'grantRequests' : grants,
+        'expectedGrantResponseCodes': [(0,), (400,), (400,)],
+    }
+    writeConfig(filename, config)
+
+
+
+  @configurable_testcase(generate_EXZ_1_default_config)
+  def test_WINNF_FT_S_EXZ_1(self,config_filename):
+    """Injected Exclusion Zones
+
+    The response should be SUCCESS for CBSDs located outside 50meters of all 
+    Exclusion Zones.
+    The responseCode = 400 for CBSDs located within 50meters of all Exclusion Zones 
+    or inside Exclusion Zones. 
+    """
+
+    #Load Exclusion Zones
+    exz_record = json.load(
+        open(os.path.join('testcases', 'testdata', 'exz_record_0.json')))
+
+    # Inject EXZ database records
+    # TODO Need to figure out how to send freqRange
+    """
+    zone_id = self._sas_admin.InjectExZone({'record':exz_record}, \
+                     {'frequencyRange':{'lowFrequency': 3620000000.0,\
+                      'highFrequency': 3630000000.0}})
+    """
+    self._sas_admin.InjectExZone({'record': exz_record})
+
+    config = loadConfig(config_filename)
+    # Very light checking of the config file.
+    self.assertEqual(
+        len(config['registrationRequests']),
+        len(config['expectedResponseCodes']))
+
+    # Inject FCC IDs.
+    for fcc_id in config['fccIds']:
+      self._sas_admin.InjectFccId({'fccId': fcc_id})
+
+    # Inject user IDs.
+    for user_id in config['userIds']:
+      self._sas_admin.InjectUserId({'userId': user_id})
+   
+    # Register N2, N3 and N4 CBSDs.
+    device_cert = self.getCertFilename('dp_client.cert')
+    device_key = self.getCertFilename('dp_client.key')
+
+    request = {'registrationRequest': config['registrationRequests']}
+    responses = self._sas.Registration(request, device_cert, device_key)['registrationResponse']
+
+    # Check registration responses.
+    self.assertEqual(len(responses), len(config['registrationRequests']))
+    cbsd_ids = []
+    for i in range(len(responses)):
+      response = responses[i]
+      expected_response_codes = config['expectedResponseCodes'][i]
+      logging.debug('Looking at response number %d', i)
+      logging.debug('Expecting to see response code in set %s in response: %s',
+                    expected_response_codes, response)
+      self.assertIn(response['response']['responseCode'],
+                    expected_response_codes)
+      if response['response']['responseCode'] == 0:  # SUCCESS
+        self.assertTrue('cbsdId' in response)
```

Comment:
These checks are not requested by the test case. Please be careful when using copy-and-paste.

Reply:
Testcase states to validate cbsdId and grantId based on response code